src/data/dataset.py

```diff
@@ -19,88 +49,149 @@ class Dataset(object):
         root (str): the root directory of the dataset
         split ("train", "val", or "test"): the split of the dataset which
             this object represents.
-        loader (LyricsLoader or MIDILoader): the object used for reading and
-            parsing
+        loader (Loader): the object used for reading and parsing
+        metadata (Metadata): the object used for persisting metadata
         split_proportions (tuple of three numbers): the unnormalized
             (train, val, test) split.
         persist (bool): persists the train/val/test split information in csv
```

Comment:
this is a little strange. your constructor expects a split to be specified (one of "train", "val" or "test"), but it also expects the proportions for all three to be specified. i believe this is for the way you split the artists below (see my comments below). if you do end up needing it this way, could you add some comment specifying why this is necessary?

Reply:
Sure

Reply:
so which one will you be doing? :) i don't see changes

Reply:
I'll add a comment. :) Haven't had a chance yet.